Improve i18n developer documentation

d-alleyne opened the pull request:
#### Proposed Changes

Improve the developer documentation by including topics that are specific to internationalization (i18n). Topics such as:

- String freeze
- CSS for RTL
- RTL issues in general
- Designing for i18n

#### Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

Review for errors.

#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to 480-gh-Automattic/i18n-issues

d-alleyne committed:
Add information on using the 'String Freeze' label

d-alleyne committed:
Add documentation for building multi-directional layouts

dlind1 commented on docs/i18n.md:
```diff
@@ -3,3 +3,66 @@
 To make text translatable, you’d use the `translate()` function from [i18n-calypso](../packages/i18n-calypso/README.md). We extract the translatable texts wrapped inside the function and write a POT file that is imported to [translate.wordpress.com](https://translate.wordpress.com/).
```
```suggestion
If you add or modify text strings displayed to the users, consider if they would be available to users who don't use English. If the text strings would be displayed to international users, they should be enabled for translation. To make text translatable, you’d use the `translate()` function from [i18n-calypso](../packages/i18n-calypso/README.md). We extract the translatable texts wrapped inside the function and write a POT file that is imported to [translate.wordpress.com](https://translate.wordpress.com/).
```

dlind1 reviewed (commented):
Added a small suggestion.

d-alleyne committed:
Update docs/i18n.md

Co-authored-by: dlind1 <dlind1@users.noreply.github.com>

d-alleyne committed:
Re-phrase sentences to improve clarity and fix typos

d-alleyne committed:
Add 'Designing for Internationalization' section

d-alleyne committed:
Use correct visual hierachy

d-alleyne committed:
Add Images, Line Breaks, and Number formats

yuliyan commented on docs/i18n.md:
```diff
@@ -1,5 +1,121 @@
-To make text translatable, you’d use the `translate()` function from [i18n-calypso](../packages/i18n-calypso/README.md). We extract the translatable texts wrapped inside the function and write a POT file that is imported to [translate.wordpress.com](https://translate.wordpress.com/).
+If you add or modify text strings that will be displayed to users, consider whether they should be visible to non-English users.
+In that case, wrap the strings in the `translate()` function from [i18n-calypso](../packages/i18n-calypso/README.md).
+We extract the translatable texts wrapped inside the function and write a POT file that is exported to [translate.wordpress.com](https://translate.wordpress.com/).
```
Maybe this should link directly to the [WordPress.com](https://translate.wordpress.com/projects/wpcom/) Project?

yuliyan commented on docs/i18n.md:
```diff
@@ -1,5 +1,121 @@
+
+### String Freeze
+
+If you're launching or updating a feature, and want to make sure that all new strings are translated before merging,
```
It might be a little more clear if you extend it a little explaining that adding the string freeze label would actually import the newly introduced strings to [WordPress.com Project in GlotPress](https://translate.wordpress.com/projects/wpcom/), making them available for translation prior to merging the changes to the trunk.

d-alleyne replied:
I have added information based on our offline discussion.

yuliyan replied:
Thank you, It looks good!

yuliyan reviewed (commented):
Nice work, I really like how concise the document is!

I only have these two small remarks:

----
Edit: There seems to be some linting issues `yarn run eslint docs/i18n.md`. Could you resolve these too?

d-alleyne committed:
Link to the WordPress.com Project in GlotPress

d-alleyne committed:
Upload image to a public hosting URL

Using relative paths for the image only works when launching the documentation using `yarn start`.
Uploading the image to github's server is free.

d-alleyne committed:
Delete unused reference

The file was uploaded to github's HTTPS server, and thus accessible from any location

d-alleyne committed:
Fix linting errors

d-alleyne committed:
Implement code review suggestions

yuliyan reviewed (approved):
LGTM!

d-alleyne committed:
Fix linting error

dlind1 commented on docs/i18n.md:
```diff
@@ -1,5 +1,129 @@
+### Leave room for text expansion
+
+When text is translated from one language to another,
+the length of the text can change. If a text container is too small or has a fixed-sized,
```
Looks like a word is missing here, should it be `has a fixed size` or `has a fixed-sized width`?

d-alleyne replied:
Good catch! It should be `fixed size`.

dlind1 reviewed (commented):
Noticed one small wording issue.

d-alleyne committed:
Fix typos

dlind1 reviewed (approved):
LGTM!
